packages/react-dom/src/__tests__/ReactDOMFragmentRefs-test.js

```diff
@@ -0,0 +1,620 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails reactcore
+ */
+
+'use strict';
+
+let React;
+let ReactDOMClient;
+let act;
+let container;
+let Fragment;
+let Activity;
+
+describe('FragmentRefs', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    Fragment = React.Fragment;
+    Activity = React.unstable_Activity;
+    ReactDOMClient = require('react-dom/client');
+    act = require('internal-test-utils').act;
+    container = document.createElement('div');
+    document.body.appendChild(container);
+  });
+
+  afterEach(() => {
+    document.body.removeChild(container);
+  });
+
+  // @gate enableFragmentRefs
+  it('attaches a ref to Fragment', async () => {
+    const fragmentRef = React.createRef();
+    const root = ReactDOMClient.createRoot(container);
+
+    await act(() =>
+      root.render(
+        <div id="parent">
+          <Fragment ref={fragmentRef}>
+            <div id="child">Hi</div>
+          </Fragment>
+        </div>,
+      ),
+    );
+    expect(container.innerHTML).toEqual(
+      '<div id="parent"><div id="child">Hi</div></div>',
+    );
+
+    expect(fragmentRef.current).not.toBe(null);
+  });
+
+  // @gate enableFragmentRefs
+  it('accepts a ref callback', async () => {
+    let fragmentRef;
+    const root = ReactDOMClient.createRoot(container);
+
+    await act(() => {
+      root.render(
+        <Fragment ref={ref => (fragmentRef = ref)}>
+          <div id="child">Hi</div>
+        </Fragment>,
+      );
+    });
+
+    expect(fragmentRef.current).not.toEqual(null);
+  });
```

Comment:
Would `fragmentRef` here have a `.current`? Seems like it's not an instance of `createRef` or `useRef` so I wouldn't expect it to be "boxed" with a `.current` but instead to be a direct reference to the reference?

Reply:
In some cases such as a Fragment's `key` changing, a new `FragmentInstance` would be assigned as the `current` value. There are also benefits to Ref API consistency, including linter and compiler rules that check for `.current` access to understand Ref types and prevent things like access during render.

Reply:
Hmm - maybe I'm misunderstanding something, however for this test it seems like `fragmentRef` is a plain old JS variable and not a ref box (like the value returned from `createRef` or `useRef`) - so I wouldn't expect it to have a `.current` property, unless the `ref` value within the function on line 63 is already a boxed value (e.g. looks like `{ current: <something> }`)?